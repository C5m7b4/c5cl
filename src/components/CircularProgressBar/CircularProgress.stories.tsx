import React from 'react';
import { Story, Meta } from '@storybook/react';

import CircularProgressBar, {
  CircularProgressProps,
} from './CircularProgressBar';

export default {
  title: 'Circular Progress Bar',
  component: CircularProgressBar,
} as Meta;

const Template: Story<CircularProgressProps> = (args) => (
  <CircularProgressBar {...args} />
);

export const CircularProgressBarExample = Template.bind({});
CircularProgressBarExample.args = {
  size: 150,
  progress: 75,
  strokeWidth: 15,
  circleStrokeOne: '#d9edfe',
  showPercent: false,
};
