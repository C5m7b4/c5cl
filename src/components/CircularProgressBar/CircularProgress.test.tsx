import React from 'react';
import { render } from '@testing-library/react';

import CircularProgressBar from './CircularProgressBar';

describe('CircularProgressBar', () => {
  test('should render correctly', () => {
    const { container } = render(
      <CircularProgressBar
        progress={25}
        size={150}
        strokeWidth={15}
        circleStrokeOne={'#ff0000'}
        showPercent={false}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should with score above 50', () => {
    const { container } = render(
      <CircularProgressBar
        progress={55}
        size={150}
        strokeWidth={15}
        circleStrokeOne={'#ff0000'}
        showPercent={false}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should with score below 25', () => {
    const { container } = render(
      <CircularProgressBar
        progress={20}
        size={150}
        strokeWidth={15}
        circleStrokeOne={'#ff0000'}
        showPercent={false}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should with score above 75', () => {
    const { container } = render(
      <CircularProgressBar
        progress={80}
        size={150}
        strokeWidth={15}
        circleStrokeOne={'#ff0000'}
        showPercent={false}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should with percent', () => {
    const { container } = render(
      <CircularProgressBar
        progress={80}
        size={150}
        strokeWidth={15}
        circleStrokeOne={'#ff0000'}
        showPercent={true}
      />
    );
    expect(container).toMatchSnapshot();
  });
  test('should render with score > 100', () => {
    const { container } = render(
      <CircularProgressBar
        progress={110}
        size={150}
        strokeWidth={15}
        circleStrokeOne={'#ff0000'}
        showPercent={true}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
