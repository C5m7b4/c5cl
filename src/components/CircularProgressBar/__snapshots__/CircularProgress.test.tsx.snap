// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgressBar should render correctly 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#ff0000"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#f59b33"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="318.0862561759665"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        x="75"
        y="75"
      >
        25
      </text>
    </svg>
  </div>
</div>
`;

exports[`CircularProgressBar should render with score > 100 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#ff0000"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#47e61c"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="-42.411500823462205"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        x="75"
        y="75"
      >
        110
        %
      </text>
    </svg>
  </div>
</div>
`;

exports[`CircularProgressBar should with percent 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#ff0000"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#47e61c"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="84.82300164692441"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        x="75"
        y="75"
      >
        80
        %
      </text>
    </svg>
  </div>
</div>
`;

exports[`CircularProgressBar should with score above 50 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#ff0000"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#daf542"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="190.85175370557994"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        x="75"
        y="75"
      >
        55
      </text>
    </svg>
  </div>
</div>
`;

exports[`CircularProgressBar should with score above 75 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#ff0000"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#47e61c"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="84.82300164692441"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        x="75"
        y="75"
      >
        80
      </text>
    </svg>
  </div>
</div>
`;

exports[`CircularProgressBar should with score below 25 1`] = `
<div>
  <div>
    <svg
      class="circular-progress-bar"
      height="150"
      width="150"
    >
      <circle
        class="circular-progress-bar-bg"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#ff0000"
        stroke-width="15"
      />
      <circle
        class="circulare-progress-bar-circle"
        cx="75"
        cy="75"
        fill="none"
        r="67.5"
        stroke="#f00e0e"
        stroke-dasharray="424.11500823462205"
        stroke-dashoffset="339.29200658769764"
        stroke-width="15"
        style="transition: stroke-dashoffset 850ms ease-in-out, stroke 500ms linear;"
      />
      <text
        class="circular-progress-bar-percentage"
        x="75"
        y="75"
      >
        20
      </text>
    </svg>
  </div>
</div>
`;
